pkg/fanal/artifact/local/fs_test.go

```diff
@@ -236,35 +237,35 @@ func TestArtifact_Inspect(t *testing.T) {
 	}
 }
 
-// TODO: fix the logic in the first place
-//func TestBuildAbsPath(t *testing.T) {
-//	tests := []struct {
-//		name          string
-//		base          string
-//		paths         []string
-//		expectedPaths []string
-//	}{
-//		{"absolute path", "/testBase", []string{"/testPath"}, []string{"/testPath"}},
-//		{"relative path", "/testBase", []string{"testPath"}, []string{"/testBase/testPath"}},
-//		{"path have '.'", "/testBase", []string{"./testPath"}, []string{"/testBase/testPath"}},
-//		{"path have '..'", "/testBase", []string{"../testPath/"}, []string{"/testPath"}},
-//	}
-//
-//	for _, test := range tests {
-//		t.Run(test.name, func(t *testing.T) {
-//			got := buildAbsPaths(test.base, test.paths)
-//			if len(test.paths) != len(got) {
-//				t.Errorf("paths not equals, expected: %s, got: %s", test.expectedPaths, got)
-//			} else {
-//				for i, path := range test.expectedPaths {
-//					if path != got[i] {
-//						t.Errorf("paths not equals, expected: %s, got: %s", test.expectedPaths, got)
-//					}
-//				}
-//			}
-//		})
-//	}
-//}
+func TestBuildPathsToSkip(t *testing.T) {
+	tests := []struct {
+		name  string
+		os    string
+		paths []string
+		base  string
+		want  []string
+	}{
+		// linux/mac
+		{"path - abs, base - abs. Skip common prefix", "linux/mac", []string{"/foo/bar"}, "/foo", []string{"/foo/bar"}},
+		//{"path - abs, base - rel.", nil, "", nil}, - impossible to check. absPath for each PC will be different. Checked manual mode.
+		{"path - rel, base - abs", "linux/mac", []string{"bar"}, "/foo", []string{"/foo/bar"}},
+		{"path - rel, base - rel. Skip common prefix", "linux/mac", []string{"foo/bar/bar"}, "foo", []string{"foo/bar/bar"}},
+		{"path - rel with dot, base - rel. Skip common prefix", "linux/mac", []string{"./foo/bar"}, "foo", []string{"foo/bar"}},
+		{"path - rel, base - dot", "linux/mac", []string{"foo/bar"}, ".", []string{"foo/bar"}},
+		// windows
+		{"path - rel, base - rel. Skip common prefix", "windows", []string{"foo\\bar\\bar"}, "foo", []string{"foo\\bar\\bar"}},
+		{"path - rel, base - dot", "windows", []string{"foo\\bar"}, ".", []string{"foo\\bar"}},
```

Comment:
It is easier to handle slashed paths. We may want to use `filepath.ToSlash`.

```suggestion
		{"path - rel, base - dot", "windows", []string{"foo\\bar"}, ".", []string{"foo/bar"}},
```

Reply:
Done